app/models/user.rb

```diff
@@ -6,6 +6,11 @@ class User < ApplicationRecord
   has_many :commentlikes, foreign_key: 'user_id'
   has_many :plikes, through: :postlikes, source: :post
   has_many :clikes, through: :commentlikes, source: :comment
+  has_many :friendships
+  has_many :friends, -> { where(friendships: { status: 'accepted' }) }, through: :friendships
```

Comment:
What about the inverted friendship relation that would make your confirmed column make much more sense and you may need a lot less of these associations you have here.

Reply:
See this [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friend.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/) and checkout the link
https://smartfunnycool.com/friendships-in-activerecord/

Reply:
We are following "Ruby On Rails - Building Social Networks" book by "Michael Hartl" because it is more clear and detailed. So we preferred to follow that path instead of document which given on the link from pathwright.